x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_indices/search_indices.tsx

```diff
@@ -49,51 +49,65 @@ export const baseBreadcrumbs = [
 ];
 
 export const SearchIndices: React.FC = () => {
-  const { fetchIndices, onPaginate } = useActions(IndicesLogic);
-  const { meta, indices, hasNoIndices, isLoading } = useValues(IndicesLogic);
+  const { fetchIndices, onPaginate, setIsFirstRequest } = useActions(IndicesLogic);
+  const { meta, indices, hasNoIndices, isLoading, isFirstRequest } = useValues(IndicesLogic);
   const [showHiddenIndices, setShowHiddenIndices] = useState(false);
   const [searchQuery, setSearchValue] = useState('');
 
   const [calloutDismissed, setCalloutDismissed] = useLocalStorage<boolean>(
     'enterprise-search-indices-callout-dismissed',
     false
   );
+
+  useEffect(() => {
+    setIsFirstRequest();
+  }, []);
+
   useEffect(() => {
-    fetchIndices({ meta, returnHiddenIndices: showHiddenIndices, searchQuery });
+    fetchIndices({
+      meta,
+      returnHiddenIndices: showHiddenIndices,
+      searchQuery,
+    });
   }, [searchQuery, meta.page.current, showHiddenIndices]);
 
-  const pageTitle = isLoading
-    ? ''
-    : indices.length !== 0
-    ? i18n.translate('xpack.enterpriseSearch.content.searchIndices.searchIndices.pageTitle', {
-        defaultMessage: 'Elasticsearch indices',
-      })
-    : i18n.translate('xpack.enterpriseSearch.content.searchIndices.searchIndices.emptyPageTitle', {
-        defaultMessage: 'Welcome to Enterprise Search',
-      });
+  const pageTitle =
+    isFirstRequest && isLoading
```

Comment:
nitpick: instead of checking `isFirstRequest && isLoading` three times in this file, have you considered moving that logic to the `isLoading` selector in the logic file instead? Added advantage that if someone in the future needs to do this, they won't forget to add `isFirstRequest` and you can add a comment explaining why we need this.

Reply:
Since we don't have any other use for isLoading I will move them and add comments. Thanks